assets/javascripts/discourse/services/chat.js

```diff
@@ -172,12 +172,12 @@ export default Service.extend({
       }
 
       if (!this._fetchingChannels) {
```

Comment:
This whole code block was suspicious, even though it's very probably that refreshChannels won't be done before we reach next line, it's still not a clean flow.